_posts/2019-06-06-NeuralODEs.markdown

```diff
@@ -0,0 +1,388 @@
+     </p>
+     <p>
+     Exercise 12.4:
+     Applying Euler's method with step-size \(h\), we get \(\hat{y}(0) = 0\), \(\hat{y}(h) = \hat{y}(0) + h \hat{y}(0)^{1/5} = 0\), \(\hat{y}(2h) = \hat{y}(h) + h \hat{y}(h)^{1/5} =0\). Iterating, we see that it holds \(y(nh)=0\) for all \(n\geq 0\). Instead, the implicit Euler's method reads
+     
+     $$\hat{y}_{n+1} = \hat{y}_n + h \hat{y}_{n+1}^{1/5}$$
+     
+     for \(n \geq 0\) and \(\hat{y}_0 = 0\). After substituting \(\hat{y}_{n} = (C_nh)^{5/4}\) in the above relation, we only need that there exists a sequence \(C_n\) satisying the requirements.
+     </p>
+     <p>
+     Exercise 12.7: 
+     First, notice that
+     
+     $$e_{n+1} = y(x_{n+1}) - y_{n} - \frac{1}{2}h(f_{n+1} + f_n)= e_n - \frac{1}{2}h (f_{n+1}+f_n) + \int_{x_n}^{x_{n+1}} f(s,y(s))\,ds$$
+     
+     and that the second component of the RHS is the same as \(E_1\) in Exercise 1 above. Therefore the first bound follows. The last inequality is simply obtained by re-arranging the terms.  
+     </p>
+     </details>
+
+  3. Consider the following method for solving \(y' = f(y)\):
+       
+     $$y_{n+1} = y_n + h*(theta*f(y_n) + (1-theta)*f(y_{n+1}))$$
+    
+     Assuming sufficient smoothness of \(y\) and \(f\), for what value of \(0 \leq\theta\leq 1\) is the truncation error the smallest? What does this mean about the accuracy of the method?
+     <details><summary>Solution</summary>
+     <p>
+     By definition, it holds
+     
+     $$h\tau_n = y_{n+1} - y_n - h (\theta f_n + (1-\theta) f_{n+1}) = y_{n+1} - y_n - h \theta y_n' - h(1-\theta) y_{n+1}'$$
+     
+     Taylor-expanding, we get
+     
+     $$h\tau_n = y_{n} + hy_n' + h^2/2y_n'' + O(h^3) - y_n - h \theta y_n' - h(1-\theta) y_{n}' - h^2(1-\theta) y_{n}'' + O(h^3) = h^2(\theta - 1/2)y_n''+O(h^3)$$
+     
+     It follows that the truncation error is the smallest for \(\theta=1/2\). For \(\theta = 1/2\) the method has order \(2\), otherwise it has order \(1\).
+     </p>
+     </details>
+
+  4. [Colab Notebook](https://colab.research.google.com/drive/1bNg-RzZoelB3w8AUQ6mefRQuN3AdrIqX)
+     <details><summary>Solution</summary>
+     <p>
+     [Solution](https://colab.research.google.com/drive/1wTQXy2_4InQH51rEmiCtvl5Q7MiyrC4k)
+     </p>
+     </details> 
+
+<br />
+
+# 2 Numerical solution of ODEs - Part 2
+  **Motivation**: In the previous class we introduced some simple schemes to numerically solve ODEs. In order to understand which numerical scheme is more proper to apply, it is important to know and understand their different properties. For this reason, in this class, we go through some more involved schemes and their convergence analysis, as well as stability analysis. 
```

Comment:
“For this reason, in this class, we go through some more involved schemes and their convergence analysis, as well as stability analysis.” —> “For this reason, in this class, we go through some more involved schemes and analyze them in terms of convergence and stability.”

Reply:
fixed